references/classification/WildFire/torch/train.py

```diff
@@ -158,18 +158,16 @@ def main(args):
     normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
 
     train_transforms = transforms.Compose([
-        transforms.RandomResizedCrop(size=args.resize // 7, scale=(0.8, 1.0)),
+        transforms.RandomResizedCrop(size=args.resize, scale=(0.8, 1.0)),
         transforms.RandomRotation(degrees=15),
         transforms.ColorJitter(),
         transforms.RandomHorizontalFlip(),
-        transforms.CenterCrop(size=args.resize),  # Image net standards
         transforms.ToTensor(),
         normalize
     ])
 
     val_transforms = transforms.Compose([
-        transforms.Resize(size=args.resize // 7),
-        transforms.CenterCrop(size=args.resize),
+        transforms.Resize(size=args.resize),
```

Comment:
Small difference here: `RandomResizedCrop` does crop **then** resize the rop.
But here we need the CenterCrop after the Resize. The reason behind it (https://pytorch.org/docs/stable/torchvision/transforms.html#torchvision.transforms.Resize) : if you pass an int to Resize, it keeps the aspect ratio and resize the smallest side to `args.resize`. If you pass a tuple it will resize the entire image to target shape (potentially changing the aspect ratio).

On eval, you probably don't want to stretch the image, but you need them to be all of the same size to batch them. So keep the `transforms.Resize` as is but add back the CenterCrop as well to make sure the image is not stretched !

Reply:
I should resize using a tuple to keep the ratio and then do a centercrop?

Reply:
I was rather suggesting doing resize with an int (`args.resize`) then doing a `CenterCrop(arg.resize)`.
If the image is squared, CenterCrop won't do a thing, if it isn't, it will make it square by cropping

Reply:
oh yes sorry, Done !